FastSimulation/Calorimetry/interface/CalorimetryManager.h

```diff
@@ -186,5 +188,8 @@ class CalorimetryManager{
   bool useShowerLibrary;
   bool useCorrectionSL;
   FastHFShowerLibrary *theHFShowerLibrary;
+
+  KKCorrectionFactors* ecalCorrection;
```

Comment:
use a unique_ptr
it destroys itself when it gets out of scope



On Mon, Aug 24, 2015 at 10:11 AM, Knut Kiesel <notifications@github.com>
wrote:

> In FastSimulation/Calorimetry/interface/CalorimetryManager.h
> <https://github.com/cms-sw/cmssw/pull/10596#discussion_r37727503>:
>
> > @@ -186,5 +188,8 @@ class CalorimetryManager{
> >    bool useShowerLibrary;
> >    bool useCorrectionSL;
> >    FastHFShowerLibrary *theHFShowerLibrary;
> > +
> > +  KKCorrectionFactors* ecalCorrection;
>
> I would like to check if the correction should be applied. Now, I check if
> the pointer is NULL, but this is not possible with a non-pointer. Do you
> recommend an additional variable 'bool doEcalCorrection' in the
> CalorimetryManager class?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cms-sw/cmssw/pull/10596/files#r37727503>.
>